src/addresses/address-details/AddressDetails.container.tsx

```diff
@@ -42,93 +26,16 @@ export const AddressDetailsPage: FC = () => {
     brokerName?: string;
     podName?: string;
   }>();
-  const [brokerDetails, setBrokerDetails] = useState<K8sResourceKind>({});
-  const [_loading, setLoading] = useState<boolean>(true);
-  const [isFirstMount, setIsFirstMount] = useState(true);
-  const [error, setError] = useState<string>('');
-  const [isErrorModalOpen, setIsErrorModalOpen] = useState<boolean>(false);
-  const [routes] = useK8sWatchResource<K8sResourceKind[]>({
-    isList: true,
-    groupVersionKind: {
-      group: 'route.openshift.io',
-      kind: 'Route',
-      version: 'v1',
-    },
-    namespaced: true,
-  });
-
-  const k8sGetBroker = () => {
-    setLoading(true);
-    k8sGet({ model: AMQBrokerModel, name: brokerName, ns: namespace })
-      .then((broker: K8sResourceKind) => {
-        setBrokerDetails(broker);
-      })
-      .catch((e) => {
-        setError(e.message);
-      })
-      .finally(() => {
-        setLoading(false);
-      });
-  };
-
-  const handleModalToggle = () => {
-    setIsErrorModalOpen(!isErrorModalOpen);
-  };
-
-  const handleTryAgain = () => {
-    setIsErrorModalOpen(false);
-    window.location.reload();
-  };
-
-  if (isFirstMount) {
-    k8sGetBroker();
-    setIsFirstMount(false);
-  }
 
-  const podOrdinal = parseInt(podName.replace(brokerName + '-ss-', ''));
-
-  const { token, isError, isLoading, source } = useJolokiaLogin(
-    brokerDetails,
-    routes,
-    podOrdinal,
+  const { brokerCr: brokerDetails, error: error } = UseGetBrokerCR(
+    brokerName,
+    namespace,
   );
 
-  const [prevIsLoading, setPrevIsLoading] = useState(isLoading);
-  const [notify, setNotify] = useState(false);
-  if (prevIsLoading !== isLoading) {
-    if (!isLoading && source === 'api') {
-      setNotify(true);
-    }
-    setPrevIsLoading(isLoading);
-  }
-  if (notify) {
-    if (isError) {
-      setIsErrorModalOpen(true);
-    }
-    setNotify(false);
-  }
+  const podOrdinal = parseInt(podName.replace(brokerName + '-ss-', ''));
 
   return (
-    <AuthContext.Provider value={token}>
-      <Modal
-        variant={ModalVariant.small}
-        title={t('login_failed')}
-        titleIconVariant="danger"
-        isOpen={isErrorModalOpen}
-        onClose={handleModalToggle}
-        actions={[
-          <Button key="confirm" variant="primary" onClick={handleTryAgain}>
-            {t('try_again')}
-          </Button>,
-          <Button key="cancel" variant="link" onClick={handleModalToggle}>
-            {t('cancel')}
-          </Button>,
-        ]}
-      >
-        <TextContent>
-          <Text component={TextVariants.h6}>{t('login_failed_message')}</Text>
-        </TextContent>
-      </Modal>
+    <JolokiaAuthentification brokerCR={brokerDetails} podOrdinal={podOrdinal}>
```

Comment:
typo: JolokiaAuthentication ?

Reply:
Ah right! Authentification is French :sweat_smile: